WcaOnRails/app/views/competitions/_competition_form.html.erb

```diff
@@ -220,7 +220,32 @@
     <% end %>
 
     <%= f.input :currency_code, collection: Money::Currency.table.values, label_method: lambda { |c| c[:name]  }, value_method: lambda { |c| c[:iso_code] }  %>
-    <%= f.input :base_entry_fee_lowest_denomination %>
+    <%= f.input :base_entry_fee_lowest_denomination, input_html: { type: "text" } %>
+    <script>
+
+      // Setup the currency mask on the fee field
+      $(function() {
+        wca.applyCurrencyMask('init',
+                              '#competition_base_entry_fee_lowest_denomination',
+                              $('#competition_currency_code').val());
+      })
+
+      // Setup form onsubmit callback
+      $('#edit-competition-form').submit(function() {
```

Comment:
I think a more common idiom for this is actually to set up an onchange listener, and populate a hidden input with the value you want sent to the server. Then the UI component the user is actually interacting with would effectively get removed from the form, so it would not get sent over to the server. You can see this with our autocomplete component, actually. The element you interact with is not the hidden input that contains user ids:

![image](https://cloud.githubusercontent.com/assets/277474/23282606/cf062358-f9d6-11e6-9a8d-4f2a93715c5c.png)

Note how the input I've selected does not actually have a `name` field, that means it will not be included when the form is submitted. There's an earlier input with `name="competition[organizer_ids]"`, `value="1554,13545"`, and `style="display: none;"`. That's the real input that gets sent over to the server.

I think one reason people generally don't reuse the existing component as you've done here is to handle the case where the submit gets cancelled for some reason (you can imagine there's another listener on the form that cancells the submit if some criteria isn't met).